# HG changeset patch
# User David Bateman <dbateman@free.fr>
# Date 1272313577 -7200
#      Mon Apr 26 22:26:17 2010 +0200
# Node ID 3eba2cc7cbda587c2f76afca134c90275b22ba6a
# Parent  43685a0541c449b4208fd75a45bc42edb93b031f
Allow matlab style linestyles in errorbar plots

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,16 +1,18 @@
 2010-04-26  David Bateman  <dbateman@free.fr>
 
 	* plot/bar.m: Document the use of the colormap for the color of
 	the bars.
 
 	* plot/private/__errplot__.m: Allow x, box and xy errorbars with
 	different syntax.
 
+	* plot/private/__errplot__.m: Allow matlab style linestyles,
+
 2010-04-24  David Bateman  <dbateman@free.fr>
 
 	* plot/private/__patch__.m: Correct order of calling setdata and
 	setvertexdata.
 
 2010-04-25  Ben Abbott <bpabbot@mac.com>
 
 	* plot/__go_draw_figure__.m: Hidden axes may have visible children.
diff --git a/scripts/plot/private/__errplot__.m b/scripts/plot/private/__errplot__.m
--- a/scripts/plot/private/__errplot__.m
+++ b/scripts/plot/private/__errplot__.m
@@ -48,30 +48,37 @@ function h = __errplot__ (fstr, p, a1, a
       ifmt = "xyerr";
     elseif (strcmp (fmt.linestyle, "#"))
       ifmt = "box";
     elseif (strcmp (fmt.linestyle, "#~"))
       ifmt = "boxy";
     elseif (strcmp (fmt.linestyle, "#~>"))
       ifmt = "boxxy";
     else
-      print_usage ();
+      ifmt = "matlab";
     endif
 
     hg = hggroup ("parent", p);
     h = [h; hg];
     args = __add_datasource__ ("__errplot__", hg, 
                                {"x", "y", "l", "u", "xl", "xu"});
 
     if (isempty (fmt.color))
       hl = __line__ (hg, "color", __next_line_color__ ());
     else
       hl = __line__ (hg, "color", fmt.color);
     endif
 
+    if (index (ifmt, "matlab"))
+      ifmt = "yerr";
+      if (! strcmp (fmt.linestyle, "none"))
+        set (hl, "linestyle", fmt.linestyle);
+      endif
+    endif
+
     ## FIXME -- note the code below adds the errorbar data directly as
     ## ldata, etc properties of the line objects, as gnuplot can handle
     ## this.  Matlab has the errorbar part of the plot as a special line
     ## object with embedded NaNs that draws the three segments of the
     ## bar separately.  Should we duplicate Matlab's behavior and stop
     ## using the ldata, etc. properties of the line objects that are
     ## Octace specific?
 
diff --git a/scripts/plot/private/__pltopt__.m b/scripts/plot/private/__pltopt__.m
--- a/scripts/plot/private/__pltopt__.m
+++ b/scripts/plot/private/__pltopt__.m
@@ -149,16 +149,18 @@ function [options, valid] = __pltopt1__ 
   if (strcmp (caller,"__errplot__"))
     if (strncmp (opt, "#~>", 3))
       n = 3;
     elseif (strncmp (opt, "#~", 2) || strncmp (opt, "~>", 2))
       n = 2;
     elseif (strncmp (opt, "~", 1) || strncmp (opt, ">", 1) 
             || strncmp (opt, "#", 1))
       n = 1;
+    else
+      n = 0;
     endif
     options.linestyle = opt(1:n);
     opt(1:n) = [];
     have_linestyle = true;
   endif
 
   while (! isempty (opt))
     if (strncmp (opt, "--", 2) || strncmp (opt, "-.", 2))
